File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/local/VehicleDao.java
package com.dani_chuks.andeladeveloper.starwars.data.db.local;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.dani_chuks.andeladeveloper.starwars.data.models.entities.Vehicle;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface VehicleDao {
    @Query("SELECT * FROM vehicle")
    Flowable<List<Vehicle>> getAll();

    @Query("SELECT * FROM vehicle LIMIT 7")
    Flowable<List<Vehicle>> getFirstSeven();

    @Query("SELECT * FROM vehicle ORDER BY name ASC" )
    Flowable<List<Vehicle>> getAllAlphabetically();

    @Query("SELECT * FROM vehicle WHERE name = :starshipName")
    Flowable <Vehicle> getPeopleByName(String starshipName);

    @Query("SELECT * FROM vehicle WHERE url = :vehicleUrl")
    Flowable<Vehicle> getVehicleByURL(String vehicleUrl);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertVehicleList(List<Vehicle> vehicleList);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertVehicle(Vehicle vehicle);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateVehicle(Vehicle vehicle);
}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/local/dao/VehicleDao.kt
package com.dani_chuks.andeladeveloper.starwars.data.db.local.dao

import androidx.room.Dao
import androidx.room.Query
import com.dani_chuks.andeladeveloper.starwars.data.models.entities.Vehicle
import kotlinx.coroutines.flow.Flow

@Dao
abstract class VehicleDao: BaseDao<Vehicle> {
    @Query("SELECT * FROM vehicle")
    abstract suspend fun all(): List<Vehicle>

    @Query("SELECT * FROM vehicle ORDER BY name ASC")
    abstract suspend fun allAlphabetically(): List<Vehicle>

    @Query("SELECT * FROM vehicle LIMIT :size")
    abstract suspend fun getItemBySize(size: Int): List<Vehicle>

    @Query("SELECT * FROM vehicle WHERE name = :starshipName")
    abstract suspend fun getVehicleByName(starshipName: String): Vehicle

    @Query("SELECT * FROM vehicle WHERE url = :vehicleUrl")
    abstract suspend fun getVehicleByURL(vehicleUrl: String): Vehicle?

    @Query("SELECT * FROM vehicle")
    abstract fun allAsFlow(): Flow<List<Vehicle>?>

    @Query("SELECT * FROM vehicle ORDER BY name ASC")
    abstract fun allAlphabeticallyAsFlow(): Flow<List<Vehicle>?>

    @Query("SELECT * FROM vehicle LIMIT :size")
    abstract fun getItemBySizeAsFlow(size: Int): Flow<List<Vehicle>?>

    @Query("SELECT * FROM vehicle WHERE name = :name")
    abstract fun getVehicleByNameAsFlow(name: String): Flow<Vehicle?>

    @Query("SELECT * FROM vehicle WHERE url = :url")
    abstract fun getVehicleByURLAsFlow(url: String): Flow<Vehicle?>
}
